src/app/core/services/categories.service.ts

```diff
@@ -1,39 +1,37 @@
-/* eslint-disable no-console */
 import { inject, Injectable } from '@angular/core';
 import { environment } from '../../../environments/environment';
-import { HttpClient } from '@angular/common/http';
-import { catchError, map, Observable, of } from 'rxjs';
+import { map, Observable, tap } from 'rxjs';
 import { QuizCategory } from '../../shared/models/quiz-category.model';
 import { RandomizationService } from './randomization.service';
+import { CategoriesStoreService } from './categories-store.service';
+import { ApiService } from './api.service';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CategoriesService {
-  private readonly categoriesUrl = environment.categoriesUrl;
   private readonly categoriesNumber = environment.numberOfQuizCategories;
   private readonly minNumberQuestions = environment.minNumberOfQuestion;
   private readonly maxNumberQuestions = environment.maxNumberOfQuestion;
 
-  private readonly http = inject(HttpClient);
+  private readonly apiService = inject(ApiService);
   private readonly randomizationService = inject(RandomizationService);
+  private readonly categoriesStoreService = inject(CategoriesStoreService);
+
+  categories!: QuizCategory[];
 
   getRandomCategories(): Observable<QuizCategory[]> {
-    return this.fetchCategories().pipe(
+    if (this.categories && this.categories.length > 0) {
+      return this.categoriesStoreService.getCategories();
+    }
+
+    return this.apiService.fetchCategories().pipe(
       map(categories => this.randomizationService
       .getRandomItems(categories, this.categoriesNumber)
       .map(category => this.enrichCategory(category))),
-    );
-  }
-
-  private fetchCategories(): Observable<QuizCategory[]> {
-    return this.http
-    .get<{ trivia_categories: QuizCategory[] }>(this.categoriesUrl)
-    .pipe(
-      map(response => response.trivia_categories),
-      catchError((error) => {
-        console.error('Error fetching categories:', error);
-        return of([]);
+      tap((categories) => {
+        this.categoriesStoreService.addCategories(categories);
+        this.categories = categories;
```

Comment:
No need to save categories in service you can access static data from the store or use one of their caching methods
https://ngneat.github.io/elf/docs/features/requests/requests-cache/